### PR TITLE
Fixes bug 1268029 - Added a Rule to replace a signature if there is a JIT classification.

### PR DESCRIPTION
**Pull request by adngdb**
cc @luser

### CODE REVIEW

File: socorro/processor/signature_utilities.py [hunk 1]
```diff
@@ -970,3 +970,31 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
...
+    #--------------------------------------------------------------------------
+    def version(self):
+        return '1.0'
+
+    #--------------------------------------------------------------------------
+    def _predicate(self, raw_crash, raw_dumps, processed_crash, proc_meta):
+        return (
+            processed_crash.get('classifications') and
+            processed_crash.classifications.get('jit') and
+            processed_crash['classifications']['jit'].get('category')
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: you could have done `processed_crash.get('classifications', {}).get('jit', {}).get('category')` :)
or 
```python
try:
    return processed_crash.classifications.jit.category
except KeyError:
    return 
```

File: socorro/processor/signature_utilities.py [hunk 2]
```diff
@@ -970,3 +970,31 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
...
+        )
+
+    #--------------------------------------------------------------------------
+    def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
+        processor_meta.processor_notes.append(
+            'Signature replaced with a JIT Crash Category, '
+            'was: {}'.format(processed_crash.signature)
+        )
+        processed_crash.signature = "jit | {}".format(
+            processed_crash.classifications['jit']['category']
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: Is it a DotDict or a `dict`?
I for one, prefer pure python `dict` objects everywhere. I think DotDict is asking for trouble.

File: socorro/processor/signature_utilities.py [hunk 3]
```diff
@@ -970,3 +970,31 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
...
+            processed_crash.get('classifications') and
+            processed_crash.classifications.get('jit') and
+            processed_crash['classifications']['jit'].get('category')
+        )
+
+    #--------------------------------------------------------------------------
+    def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
+        processor_meta.processor_notes.append(
+            'Signature replaced with a JIT Crash Category, '
+            'was: {}'.format(processed_crash.signature)
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: Pardon my lack of experience with processors but wouldn't it be better to use `...was: "{}"'.format(...)`?

**Comment by peterbe** [comment]
Thanks for making the pep8 cleanup and nice that it's a separate commit!
As you can tell I'm against using DotDict. I base that on years of experience of weird DotDict errors that are hard to debug, the fact that you can't easily transpose them as JSON and the years of using MongoDB and trying to mess with a DotDict equivalent. Not a nit. Just a drive-by opinion. 

I do think we you might want to at least look at the nit I mentioned in the `_predicate` method. Otherwise r+

File: socorro/processor/signature_utilities.py (hunk 2 above)
Comment (adngdb) [inline socorro/processor/signature_utilities.py]: ``processed_crash`` I know for sure is a DotDict, but I think ``classifications`` is not. Hence the syntax. But you're right, and I agree about DotDict being asking for trouble, even though it's a nicer syntax.

**Pull request merged**